core/server/data/migrations/versions/4.6/01-remove-comped-status.js

```diff
@@ -0,0 +1,39 @@
+const logging = require('../../../../../shared/logging');
+const {createTransactionalMigration} = require('../../utils.js');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Updating all member "comped" statuses to "paid"');
+        await knex('members')
+            .update('status', 'paid')
+            .where('status', 'comped');
+    },
+    async function down(knex) {
+        const compedMemberIds = (await knex('members')
+            .select('members.id')
+            .innerJoin(
+                'members_stripe_customers',
+                'members.id',
+                'members_stripe_customers.member_id'
+            ).innerJoin(
+                'members_stripe_customers_subscriptions',
+                function () {
+                    this.on(
+                        'members_stripe_customers.customer_id',
+                        'members_stripe_customers_subscriptions.customer_id'
+                    ).onIn(
+                        'members_stripe_customers_subscriptions.status',
+                        ['active', 'trialing', 'past_due', 'unpaid']
+                    );
+                }
+            ).where(
+                'members_stripe_customers_subscriptions.plan_nickname',
+                '=',
+                'Complimentary'
+            )).map(({id}) => id);
+
```

Comment:
Adding a log message here saying how many members are being updated would be useful. Also, do we even want to run a query if `compedMemberIds.length === 0` ?

Reply:
Done :+1: